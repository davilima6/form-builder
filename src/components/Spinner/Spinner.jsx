// @flow
import * as React from 'react';
import styled, { css } from 'styled-components';
import spinner from './spinner.svg';

const SpinnerWrapper = styled.div`
  text-align: center;
`;

const StyledSpinner = styled.img`
  ${props => props.size
    && css`
      height: ${props.size}px;
      width: ${props.size}px;
    `}
`;

type Props = {
  size?: number,
};

const Spinner = ({ size }: Props) => (
  <SpinnerWrapper>
    <StyledSpinner size={size} src={spinner} alt="Loading..." />
  </SpinnerWrapper>
);

Spinner.defaultProps = {
  size: 200,
};

export default Spinner;
