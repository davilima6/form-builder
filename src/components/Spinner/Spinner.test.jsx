import React from 'react';
import { cleanup, render } from 'react-testing-library';
import 'jest-dom/extend-expect';
// import 'jest-styled-components';
import Spinner from './Spinner';

describe('Spinner', () => {
  let component;

  beforeEach(() => {
    cleanup();
    component = render(<Spinner />);
  });

  test('renders with no props without crashing', () => {
    expect(component).toBeDefined();
  });

  test('matches snapshot', () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
  });
});
