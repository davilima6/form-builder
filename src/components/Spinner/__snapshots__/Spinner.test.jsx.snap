// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa khUSez"
  >
    <img
      alt="Loading..."
      height="200"
      src="spinner.svg"
      width="200"
    />
  </div>
</DocumentFragment>
`;
